make fails with GNU 10 compilers

Good morning,
I am compiling the git master version of MCT on a Mac Os X (Mojave) under Brew.
Brew has recently updated gnu compilers to version 10 (as Ubuntu has done on the linux side since version 20.10).

Polymorphic fortran API's are not accepted anymore by default and they require the
`-fallow-argument-mismatch`
option for gfortran (or mpifort, mpif90 and similar wrappers).

The configure should be able to detect the gnu compiler version and add this option (unfortunately not recognised by previous gnu compilers, therefore the test on the version is needed), otherwise make fails on
```
mpicc -c -DSYSDARWIN -DCPRINTEL -g -O2  -I. -I../ get_zeits.c
mpif90 -c  -I. -I../ -DSYSDARWIN -DCPRINTEL -O2  -g -O2   m_mpif.F90
mpif90 -c  -I. -I../ -DSYSDARWIN -DCPRINTEL -O2  -g -O2   m_realkinds.F90
mpif90 -c  -I. -I../ -DSYSDARWIN -DCPRINTEL -O2  -g -O2   m_stdio.F90
mpif90 -c  -I. -I../ -DSYSDARWIN -DCPRINTEL -O2  -g -O2   m_mpif90.F90
mpif90 -c  -I. -I../ -DSYSDARWIN -DCPRINTEL -O2  -g -O2   m_dropdead.F90
mpif90 -c  -I. -I../ -DSYSDARWIN -DCPRINTEL -O2  -g -O2   m_chars.F90
mpif90 -c  -I. -I../ -DSYSDARWIN -DCPRINTEL -O2  -g -O2   m_flow.F90
mpif90 -c  -I. -I../ -DSYSDARWIN -DCPRINTEL -O2  -g -O2   m_ioutil.F90
mpif90 -c  -I. -I../ -DSYSDARWIN -DCPRINTEL -O2  -g -O2   m_mpout.F90
mpif90 -c  -I. -I../ -DSYSDARWIN -DCPRINTEL -O2  -g -O2   m_die.F90
mpif90 -c  -I. -I../ -DSYSDARWIN -DCPRINTEL -O2  -g -O2   m_IndexBin_char.F90
mpif90 -c  -I. -I../ -DSYSDARWIN -DCPRINTEL -O2  -g -O2   m_IndexBin_integer.F90
mpif90 -c  -I. -I../ -DSYSDARWIN -DCPRINTEL -O2  -g -O2   m_IndexBin_logical.F90
mpif90 -c  -I. -I../ -DSYSDARWIN -DCPRINTEL -O2  -g -O2   m_mall.F90
mpif90 -c  -I. -I../ -DSYSDARWIN -DCPRINTEL -O2  -g -O2   m_String.F90
m_String.F90:478:17:

  457 |   call MPI_bcast(ln,1,MP_INTEGER,root,comm,ier)
      |                 2
......
  478 |   call MPI_bcast(Str%c(1),ln,MP_CHARACTER,root,comm,ier)
      |                 1
Error: Type mismatch between actual argument at (1) and actual argument at (2) (CHARACTER(1)/INTEGER(4)).
make[1]: *** [m_String.o] Error 1
make: *** [subdirs] Error 2
```

Thank you very much
A.P.

--- a/configure
+++ b/configure
@@ -5011,7 +5011,7 @@ if echo $FC | grep xlf >/dev/null 2>&1; then
    if test "$FCFLAGS" = "$DEFFCFLAGS"; then
        FCFLAGS=""
    fi
-elif echo $FC | grep pgf >/dev/null 2>&1; then
+elif echo $ac_fc_version_output | grep -i pgfor >/dev/null 2>&1; then
    echo "Fortran Compiler is Portland Group"
    CPRDEF="PGI"
    if test -z "$REAL8"; then
@@ -5033,6 +5033,7 @@ elif echo $FC | grep pgf >/dev/null 2>&1; then
       DEBUG="-g"
    fi
 elif test "$IS_GNU" = "yes" ; then
+# The GNU compiler can hide inside other compilers like ftn
   if echo $ac_fc_version_output | grep -i Cray >/dev/null 2>&1; then
    echo "Fortran Compiler is GNU, Cray"
     CPRDEF="GNU"
@@ -5052,7 +5053,7 @@ elif test "$IS_GNU" = "yes" ; then
     fi
 #  Put ifort after gnu in case wrapper script mpifort
 #  from openmpi is wrapping gnu.  Prevents misidentification
-elif echo $FC | grep ifort >/dev/null 2>&1; then
+elif echo $ac_fc_version_output | grep -i 'intel.*fortran' >/dev/null 2>&1; then
    echo "Fortran Compiler is Intel ifort"
    CPRDEF="INTEL"
    if test -z "$REAL8"; then
@@ -5073,6 +5074,7 @@ elif echo $FC | grep ifort >/dev/null 2>&1; then
    if test -z "$DEBUG"; then
       DEBUG="-g"
    fi
+# Put ftn after GNU for the case where ftn is wrapping PGF
 elif echo $FC | grep ftn >/dev/null 2>&1; then
  if echo $ac_fc_version_output | grep -i Portland >/dev/null 2>&1; then
    echo "Fortran Compiler is Portland Group, Cray"

--- a/configure
+++ b/configure
@@ -3681,6 +3681,12 @@ ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 
+if test -z "$GFC"; then
+   IS_GNU="no"
+else
+   IS_GNU=$GFC
+fi
+
 # CHECK FOR MPI LIBRARIES
 ac_ext=${ac_fc_srcext-f}
 ac_compile='$FC -c $FCFLAGS $ac_fcflags_srcext conftest.$ac_ext >&5'
@@ -4067,7 +4073,7 @@ _ACEOF
 # 1 to this macro) to the Fortran 90 compiler in order to get "version" output
 ac_save_FCFLAGS=$FCFLAGS
 FCFLAGS="$FCFLAGS $ac_version"
-(eval echo $as_me:4070: \"$ac_link\") >&5
+(eval echo $as_me:4076: \"$ac_link\") >&5
 ac_fc_version_output=`eval $ac_link 5>&1 2>&1 | grep -v 'Driving:'`
 echo "$ac_fc_version_output" >&5
 FCFLAGS=$ac_save_FCFLAGS
@@ -5050,6 +5056,26 @@ elif echo $FC | grep ftn >/dev/null 2>&1; then
       DEBUG="-g"
    fi
  fi
+elif test "$IS_GNU" = "yes" ; then
+  if echo $FC | grep g95 >/dev/null 2>&1; then
+    echo "Fortran Compiler is GNU g95"
+    CPRDEF="GNU"
+  else
+    echo "Fortran Compiler is GNU"
+    CPRDEF="GNU"
+# For gfortran, default flags are different
+    if test "$FCFLAGS" = "-g -O2"; then
+       FCFLAGS=""
+    fi
+    if test -z "$DEBUG"; then
+       DEBUG="-g"
+    fi
+    if test -z "$OPT"; then
+       OPT="-O2"
+    fi
+  fi
+#  Put ifort after gnu in case wrapper script mpifort
+#  from openmpi is wrapping gnu.  Prevents misidentification
 elif echo $FC | grep ifort >/dev/null 2>&1; then
    echo "Fortran Compiler is Intel ifort"
    CPRDEF="INTEL"
@@ -5070,22 +5096,6 @@ elif echo $FC | grep ifort >/dev/null 2>&1; then
    fi
    if test -z "$DEBUG"; then
       DEBUG="-g"
-   fi
-elif echo $FC | grep g95 >/dev/null 2>&1; then
-   echo "Fortran Compiler is GNU"
-   CPRDEF="GNU"
-elif echo $FC | grep gfortran >/dev/null 2>&1; then
-   echo "Fortran Compiler is GNU"
-   CPRDEF="GNU"
-# For gfortran, default flags are different
-   if test "$FCFLAGS" = "-g -O2"; then
-      FCFLAGS=""
-   fi
-   if test -z "$DEBUG"; then
-      DEBUG="-g"
-   fi
-   if test -z "$OPT"; then
-      OPT="-O2"
    fi
 elif echo $ac_fc_version_output | grep -i nag >/dev/null 2>&1; then
    echo "Fortran Compiler is NAG"
@@ -5233,6 +5243,64 @@ elif echo $ac_fc_version_output | grep -i sx >/dev/null 2>&1; then
       OPT="-Chopt"
    fi
 fi
+
+# Test to see if fortran compiler supports the flag
+# -fallow-argument-mismatch flag introduced in gfortran 10.
+#
+# Also allow support for NAG compiler using the -mismatch_all flag.
+#
+# See https://github.com/Unidata/netcdf-fortran/issues/212
+# See https://github.com/Unidata/netcdf-fortran/issues/218
+ac_save_FCFLAGS="$FCFLAGS"
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking if Fortran compiler supports allow-mismatch flag" >&5
+$as_echo_n "checking if Fortran compiler supports allow-mismatch flag... " >&6; }
+cat <<EOF >conftest.f90
+Program test
+USE ISO_C_BINDING, ONLY: C_PTRDIFF_T
+End Program
+EOF
+doit='$FC -c ${FCFLAGS} ${FCFLAGS_f90} -fallow-argument-mismatch conftest.f90'
+if { { eval echo "\"\$as_me\":${as_lineno-$LINENO}: \"$doit\""; } >&5
+  (eval $doit) 2>&5
+  ac_status=$?
+  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
+  test $ac_status = 0; }; then
+   nf_allow_mismatch=yes
+   FCFLAGS="${FCFLAGS} -fallow-argument-mismatch"
+  else
+   nf_allow_mismatch=no
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $nf_allow_mismatch" >&5
+$as_echo "$nf_allow_mismatch" >&6; }
+# End testing of gfortan allow-mismatch flags.
+
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking if Fortran compiler supports mismatch_all flag" >&5
+$as_echo_n "checking if Fortran compiler supports mismatch_all flag... " >&6; }
+cat <<EOF >conftest.f90
+Program test
+USE ISO_C_BINDING, ONLY: C_PTRDIFF_T
+End Program
+EOF
+doit='$FC -c ${FCFLAGS} ${FCFLAGS_f90} -mismatch_all conftest.f90'
+if { { eval echo "\"\$as_me\":${as_lineno-$LINENO}: \"$doit\""; } >&5
+  (eval $doit) 2>&5
+  ac_status=$?
+  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
+  test $ac_status = 0; }; then
+   nf_mismatch_all=yes
+   FCFLAGS="${FCFLAGS} -mismatch_all"
+  else
+   nf_mismatch_all=no
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $nf_mismatch_all" >&5
+$as_echo "$nf_mismatch_all" >&6; }
+
+#end testing of NAG mismatch_all flag.
+
+##
+# End mismatch checks
+##
+
 
 ###########################################################
 # END of compiler-specific flag setting

--- a/configure
+++ b/configure
@@ -3474,7 +3474,7 @@ ac_compile='$FC -c $FCFLAGS $ac_fcflags_srcext conftest.$ac_ext >&5'
 ac_link='$FC -o conftest$ac_exeext $FCFLAGS $LDFLAGS $ac_fcflags_srcext conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_fc_compiler_gnu
 if test -n "$ac_tool_prefix"; then
-  for ac_prog in nagfor xlf95 pgf95 ifort gfortran pathf95 ftn lf95 f95 fort ifc efc g95 xlf90 pgf90 pathf90 epcf90 pghpf
+  for ac_prog in nagfor xlf95 pgf95 ifort gfortran pathf95 ftn lf95 f95 fort ifc efc xlf90 pgf90 pathf90 epcf90 pghpf
   do
     # Extract the first word of "$ac_tool_prefix$ac_prog", so it can be a program name with args.
 set dummy $ac_tool_prefix$ac_prog; ac_word=$2
@@ -3518,7 +3518,7 @@ fi
 fi
 if test -z "$FC"; then
   ac_ct_FC=$FC
-  for ac_prog in nagfor xlf95 pgf95 ifort gfortran pathf95 ftn lf95 f95 fort ifc efc g95 xlf90 pgf90 pathf90 epcf90 pghpf
+  for ac_prog in nagfor xlf95 pgf95 ifort gfortran pathf95 ftn lf95 f95 fort ifc efc xlf90 pgf90 pathf90 epcf90 pghpf
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
@@ -5032,6 +5032,47 @@ elif echo $FC | grep pgf >/dev/null 2>&1; then
    if test -z "$DEBUG"; then
       DEBUG="-g"
    fi
+elif test "$IS_GNU" = "yes" ; then
+  if echo $ac_fc_version_output | grep -i Cray >/dev/null 2>&1; then
+   echo "Fortran Compiler is GNU, Cray"
+    CPRDEF="GNU"
+  else
+    echo "Fortran Compiler is GNU"
+    CPRDEF="GNU"
+  fi
+# For gfortran, default flags are different
+    if test "$FCFLAGS" = "-g -O2"; then
+       FCFLAGS=""
+    fi
+    if test -z "$DEBUG"; then
+       DEBUG="-g"
+    fi
+    if test -z "$OPT"; then
+       OPT="-O2"
+    fi
+#  Put ifort after gnu in case wrapper script mpifort
+#  from openmpi is wrapping gnu.  Prevents misidentification
+elif echo $FC | grep ifort >/dev/null 2>&1; then
+   echo "Fortran Compiler is Intel ifort"
+   CPRDEF="INTEL"
+   if test -z "$REAL8"; then
+      REAL8="-r8"
+   fi
+   if test "$FCFLAGS" = "$DEFFCFLAGS"; then
+      FCFLAGS="-w -ftz"
+   fi
+   if test -z "$PROGFCFLAGS"; then
+      PROGFCFLAGS="-assume byterecl"
+   fi
+   if test -z "$ENDIAN"; then
+      ENDIAN="-convert big_endian"
+   fi
+   if test -z "$OPT"; then
+      OPT="-O2"
+   fi
+   if test -z "$DEBUG"; then
+      DEBUG="-g"
+   fi
 elif echo $FC | grep ftn >/dev/null 2>&1; then
  if echo $ac_fc_version_output | grep -i Portland >/dev/null 2>&1; then
    echo "Fortran Compiler is Portland Group, Cray"
@@ -5056,47 +5097,6 @@ elif echo $FC | grep ftn >/dev/null 2>&1; then
       DEBUG="-g"
    fi
  fi
-elif test "$IS_GNU" = "yes" ; then
-  if echo $FC | grep g95 >/dev/null 2>&1; then
-    echo "Fortran Compiler is GNU g95"
-    CPRDEF="GNU"
-  else
-    echo "Fortran Compiler is GNU"
-    CPRDEF="GNU"
-# For gfortran, default flags are different
-    if test "$FCFLAGS" = "-g -O2"; then
-       FCFLAGS=""
-    fi
-    if test -z "$DEBUG"; then
-       DEBUG="-g"
-    fi
-    if test -z "$OPT"; then
-       OPT="-O2"
-    fi
-  fi
-#  Put ifort after gnu in case wrapper script mpifort
-#  from openmpi is wrapping gnu.  Prevents misidentification
-elif echo $FC | grep ifort >/dev/null 2>&1; then
-   echo "Fortran Compiler is Intel ifort"
-   CPRDEF="INTEL"
-   if test -z "$REAL8"; then
-      REAL8="-r8"
-   fi
-   if test "$FCFLAGS" = "$DEFFCFLAGS"; then
-      FCFLAGS="-w -ftz"
-   fi
-   if test -z "$PROGFCFLAGS"; then
-      PROGFCFLAGS="-assume byterecl"
-   fi
-   if test -z "$ENDIAN"; then
-      ENDIAN="-convert big_endian"
-   fi
-   if test -z "$OPT"; then
-      OPT="-O2"
-   fi
-   if test -z "$DEBUG"; then
-      DEBUG="-g"
-   fi
 elif echo $ac_fc_version_output | grep -i nag >/dev/null 2>&1; then
    echo "Fortran Compiler is NAG"
    CPRDEF="NAG"

--- a/configure.ac
+++ b/configure.ac
@@ -314,7 +314,7 @@ if echo $FC | grep xlf >/dev/null 2>&1; then
    if test "$FCFLAGS" = "$DEFFCFLAGS"; then
        FCFLAGS=""
    fi
-elif echo $FC | grep pgf >/dev/null 2>&1; then
+elif echo $ac_fc_version_output | grep -i pgfor >/dev/null 2>&1; then
    echo "Fortran Compiler is Portland Group"
    CPRDEF="PGI"
    if test -z "$REAL8"; then
@@ -356,7 +356,7 @@ elif test "$IS_GNU" = "yes" ; then
     fi
 #  Put ifort after gnu in case wrapper script mpifort
 #  from openmpi is wrapping gnu.  Prevents misidentification
-elif echo $FC | grep ifort >/dev/null 2>&1; then
+elif echo $ac_fc_version_output | grep -i 'intel.*fortran' >/dev/null 2>&1; then
    echo "Fortran Compiler is Intel ifort"
    CPRDEF="INTEL"
    if test -z "$REAL8"; then

--- a/configure.ac
+++ b/configure.ac
@@ -143,7 +143,7 @@ AC_C_BIGENDIAN
 
 # CHECK FOR THE FORTRAN COMPILER
 # RLJ- specify the order, include PathScale and do not search for F77
-AC_PROG_FC([nagfor xlf95 pgf95 ifort gfortran pathf95 ftn lf95 f95 fort ifc efc g95 xlf90 pgf90 pathf90 epcf90 pghpf])
+AC_PROG_FC([nagfor xlf95 pgf95 ifort gfortran pathf95 ftn lf95 f95 fort ifc efc xlf90 pgf90 pathf90 epcf90 pghpf])
 
 if test -z "$GFC"; then
    IS_GNU="no"
@@ -335,6 +335,49 @@ elif echo $FC | grep pgf >/dev/null 2>&1; then
    if test -z "$DEBUG"; then
       DEBUG="-g"
    fi
+elif test "$IS_GNU" = "yes" ; then
+# The GNU compiler can hide inside other compilers like ftn
+  if echo $ac_fc_version_output | grep -i Cray >/dev/null 2>&1; then
+   echo "Fortran Compiler is GNU, Cray"
+    CPRDEF="GNU"
+  else
+    echo "Fortran Compiler is GNU"
+    CPRDEF="GNU"
+  fi
+# For gfortran, default flags are different
+    if test "$FCFLAGS" = "-g -O2"; then
+       FCFLAGS=""
+    fi
+    if test -z "$DEBUG"; then
+       DEBUG="-g"
+    fi
+    if test -z "$OPT"; then
+       OPT="-O2"
+    fi
+#  Put ifort after gnu in case wrapper script mpifort
+#  from openmpi is wrapping gnu.  Prevents misidentification
+elif echo $FC | grep ifort >/dev/null 2>&1; then
+   echo "Fortran Compiler is Intel ifort"
+   CPRDEF="INTEL"
+   if test -z "$REAL8"; then
+      REAL8="-r8"
+   fi
+   if test "$FCFLAGS" = "$DEFFCFLAGS"; then
+      FCFLAGS="-w -ftz"
+   fi
+   if test -z "$PROGFCFLAGS"; then
+      PROGFCFLAGS="-assume byterecl"
+   fi
+   if test -z "$ENDIAN"; then
+      ENDIAN="-convert big_endian"
+   fi
+   if test -z "$OPT"; then
+      OPT="-O2"
+   fi
+   if test -z "$DEBUG"; then
+      DEBUG="-g"
+   fi
+# Put ftn after GNU for the case where ftn is wrapping PGF
 elif echo $FC | grep ftn >/dev/null 2>&1; then
  if echo $ac_fc_version_output | grep -i Portland >/dev/null 2>&1; then
    echo "Fortran Compiler is Portland Group, Cray"
@@ -359,47 +402,6 @@ elif echo $FC | grep ftn >/dev/null 2>&1; then
       DEBUG="-g"
    fi
  fi
-elif test "$IS_GNU" = "yes" ; then
-  if echo $FC | grep g95 >/dev/null 2>&1; then
-    echo "Fortran Compiler is GNU g95"
-    CPRDEF="GNU"
-  else
-    echo "Fortran Compiler is GNU"
-    CPRDEF="GNU"
-# For gfortran, default flags are different
-    if test "$FCFLAGS" = "-g -O2"; then
-       FCFLAGS=""
-    fi
-    if test -z "$DEBUG"; then
-       DEBUG="-g"
-    fi
-    if test -z "$OPT"; then
-       OPT="-O2"
-    fi
-  fi
-#  Put ifort after gnu in case wrapper script mpifort
-#  from openmpi is wrapping gnu.  Prevents misidentification
-elif echo $FC | grep ifort >/dev/null 2>&1; then
-   echo "Fortran Compiler is Intel ifort"
-   CPRDEF="INTEL"
-   if test -z "$REAL8"; then
-      REAL8="-r8"
-   fi
-   if test "$FCFLAGS" = "$DEFFCFLAGS"; then
-      FCFLAGS="-w -ftz"
-   fi
-   if test -z "$PROGFCFLAGS"; then
-      PROGFCFLAGS="-assume byterecl"
-   fi
-   if test -z "$ENDIAN"; then
-      ENDIAN="-convert big_endian"
-   fi
-   if test -z "$OPT"; then
-      OPT="-O2"
-   fi
-   if test -z "$DEBUG"; then
-      DEBUG="-g"
-   fi
 elif echo $ac_fc_version_output | grep -i nag >/dev/null 2>&1; then
    echo "Fortran Compiler is NAG"
    CPRDEF="NAG"

--- a/configure.ac
+++ b/configure.ac
@@ -145,6 +145,12 @@ AC_C_BIGENDIAN
 # RLJ- specify the order, include PathScale and do not search for F77
 AC_PROG_FC([nagfor xlf95 pgf95 ifort gfortran pathf95 ftn lf95 f95 fort ifc efc g95 xlf90 pgf90 pathf90 epcf90 pghpf])
 
+if test -z "$GFC"; then
+   IS_GNU="no"
+else
+   IS_GNU=$GFC
+fi
+
 # CHECK FOR MPI LIBRARIES
 AC_LANG_PUSH(Fortran)
 
@@ -353,6 +359,26 @@ elif echo $FC | grep ftn >/dev/null 2>&1; then
       DEBUG="-g"
    fi
  fi
+elif test "$IS_GNU" = "yes" ; then
+  if echo $FC | grep g95 >/dev/null 2>&1; then
+    echo "Fortran Compiler is GNU g95"
+    CPRDEF="GNU"
+  else
+    echo "Fortran Compiler is GNU"
+    CPRDEF="GNU"
+# For gfortran, default flags are different
+    if test "$FCFLAGS" = "-g -O2"; then
+       FCFLAGS=""
+    fi
+    if test -z "$DEBUG"; then
+       DEBUG="-g"
+    fi
+    if test -z "$OPT"; then
+       OPT="-O2"
+    fi
+  fi
+#  Put ifort after gnu in case wrapper script mpifort
+#  from openmpi is wrapping gnu.  Prevents misidentification
 elif echo $FC | grep ifort >/dev/null 2>&1; then
    echo "Fortran Compiler is Intel ifort"
    CPRDEF="INTEL"
@@ -373,22 +399,6 @@ elif echo $FC | grep ifort >/dev/null 2>&1; then
    fi
    if test -z "$DEBUG"; then
       DEBUG="-g"
-   fi
-elif echo $FC | grep g95 >/dev/null 2>&1; then
-   echo "Fortran Compiler is GNU"
-   CPRDEF="GNU"
-elif echo $FC | grep gfortran >/dev/null 2>&1; then
-   echo "Fortran Compiler is GNU"
-   CPRDEF="GNU"
-# For gfortran, default flags are different
-   if test "$FCFLAGS" = "-g -O2"; then
-      FCFLAGS=""
-   fi
-   if test -z "$DEBUG"; then
-      DEBUG="-g"
-   fi
-   if test -z "$OPT"; then
-      OPT="-O2"
    fi
 elif echo $ac_fc_version_output | grep -i nag >/dev/null 2>&1; then
    echo "Fortran Compiler is NAG"
@@ -536,6 +546,52 @@ elif echo $ac_fc_version_output | grep -i sx >/dev/null 2>&1; then
       OPT="-Chopt"
    fi
 fi
+
+# Test to see if fortran compiler supports the flag
+# -fallow-argument-mismatch flag introduced in gfortran 10.
+#
+# Also allow support for NAG compiler using the -mismatch_all flag.
+#
+# See https://github.com/Unidata/netcdf-fortran/issues/212
+# See https://github.com/Unidata/netcdf-fortran/issues/218
+ac_save_FCFLAGS="$FCFLAGS"
+AC_MSG_CHECKING([if Fortran compiler supports allow-mismatch flag])
+cat <<EOF >conftest.f90
+Program test
+USE ISO_C_BINDING, ONLY: C_PTRDIFF_T
+End Program
+EOF
+doit='$FC -c ${FCFLAGS} ${FCFLAGS_f90} -fallow-argument-mismatch conftest.f90'
+if AC_TRY_EVAL(doit); then
+   nf_allow_mismatch=yes
+   FCFLAGS="${FCFLAGS} -fallow-argument-mismatch"
+  else
+   nf_allow_mismatch=no
+fi
+AC_MSG_RESULT([$nf_allow_mismatch])
+# End testing of gfortan allow-mismatch flags.
+
+AC_MSG_CHECKING([if Fortran compiler supports mismatch_all flag])
+cat <<EOF >conftest.f90
+Program test
+USE ISO_C_BINDING, ONLY: C_PTRDIFF_T
+End Program
+EOF
+doit='$FC -c ${FCFLAGS} ${FCFLAGS_f90} -mismatch_all conftest.f90'
+if AC_TRY_EVAL(doit); then
+   nf_mismatch_all=yes
+   FCFLAGS="${FCFLAGS} -mismatch_all"
+  else
+   nf_mismatch_all=no
+fi
+AC_MSG_RESULT([$nf_mismatch_all])
+
+#end testing of NAG mismatch_all flag.
+
+##
+# End mismatch checks
+##
+
 
 ###########################################################
 # END of compiler-specific flag setting

